pallets/pool-fees/src/lib.rs

```diff
@@ -0,0 +1,840 @@
+		PositiveNAVTooOld,
+		/// Only the PoolAdmin can execute a given operation.
+		NotPoolAdmin,
+		/// The pool bucket has reached the maximum fees size.
+		MaxPoolFeesPerBucket,
+		/// The change id does not belong to a pool fees change.
+		ChangeIdNotPoolFees,
+		/// The change id belongs to a pool fees change but was called in the
+		/// wrong context.
+		ChangeIdUnrelated,
+		/// The fee can only be charged by the destination
+		UnauthorizedCharge,
+		/// The fee can only be edited or removed by the editor
+		UnauthorizedEdit,
+		/// Attempted to charge a fee of unchargeable type
+		CannotBeCharged,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Propose to append a new fee to the given (pool, bucket) pair.
+		///
+		/// Origin must be by pool admin.
+		#[pallet::call_index(0)]
+		#[pallet::weight(Weight::from_parts(10_000, 0))]
+		pub fn propose_new_fee(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			bucket: PoolFeeBucket,
+			fee: PoolFeeOf<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+			ensure!(
+				T::Permissions::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::PoolAdmin)
+				),
+				Error::<T>::NotPoolAdmin
+			);
+
+			T::ChangeGuard::note(
+				pool_id,
+				Change::AppendFee(bucket.clone(), fee.clone()).into(),
+			)?;
+
+			Self::deposit_event(Event::<T>::Proposed {
+				pool_id,
+				bucket,
+				fee,
+			});
+
+			Ok(())
+		}
+
+		/// Execute a successful fee append proposal for the given (pool,
+		/// bucket) pair.
+		///
+		/// Origin unrestriced due to pre-check via proposal gate.
+		#[pallet::call_index(1)]
+		#[pallet::weight(Weight::from_parts(10_000, 0))]
+		pub fn apply_new_fee(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			change_id: T::Hash,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			ensure!(
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+			let (bucket, fee) = Self::get_released_change(pool_id, change_id)
+				.map(|Change::AppendFee(bucket, fee)| (bucket, fee))?;
+
+			Self::add_fee(pool_id, bucket, fee)?;
+
+			Ok(())
+		}
+
+		/// Remove a fee.
+		///
+		/// Origin must be the fee editor.
+		#[pallet::call_index(2)]
+		#[pallet::weight(Weight::from_parts(10_000, 0))]
+		pub fn remove_fee(origin: OriginFor<T>, fee_id: T::FeeId) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			FeeIdsToPoolBucket::<T>::get(fee_id)
+				.map(|(pool_id, bucket)| {
+					ActiveFees::<T>::get(pool_id, bucket)
+						.into_iter()
+						.find(|fee| fee.id == fee_id)
+				})
+				.flatten()
+				.map(|fee| {
+					ensure!(
+						fee.editor.matches_account(&who),
+						Error::<T>::UnauthorizedEdit
+					);
+					Ok::<_, DispatchError>(())
+				})
+				.transpose()?
+				.ok_or(Error::<T>::FeeNotFound)?;
+
+			Self::do_remove_fee(fee_id)?;
```

Comment:
Thanks for raising that concern. I had a `TODO` for that question but decided to drop it because that should entirely be up to the fee editor AFAICT. Removing a fee does not negatively impact the pool as it increases the amount of usable funds. WDYT @mustermeiszer ?

Reply:
Mmm.. makes sense then to not put it through `ChangeGuard`.

If the change does not affect the investor, then it should not be performed over `ChangeGuard`, which I understand, that's the case

Reply:
Removing is fine I think!